### ISSUE TITLE
Using alternative compiler on windows

### ISSUE THREAD

**Issue by wmjb** [pypa/distutils#321]
I would like to see the ability to use alternative compilers on windows, other than msvc. for example using llvm-mingw as a compiler.

**Comment by wmjb** [comment]
a temporary workaround seems to be comment out the check https://github.com/pypa/distutils/blob/49e362ce78427d51b104664d246296f7546176a1/distutils/ccompiler.py#L1086
and it compiles with mingw

**Comment by Avasam** [comment]
Looking at https://github.com/pypa/distutils/blob/d636f5f8a6a5433ca04326a759aaec981af16dfa/distutils/util.py#L57
If you set your env var `VSCMD_ARG_TGT_ARCH` to anything else than `arm64`, `arm`, `x64`, `x86` (could be empty). Then `get_platform` returns the same as `sysconfig.get_platform()`, which itself should return something like `mingw_x86_64`, `mingw_i686`, `mingw_aarch64`, ... if you're running a Python compiled with mingw (for example by using MSYS2, which sets up a lot for you).

In other words, use MSYS2: https://www.mingw-w64.org/getting-started/msys2/

Or hackily highjack the compiler selection like pywin32 does: https://github.com/mhammond/pywin32/blob/1e89f41c6a98bba7ed23075322b9fa6a9bfc0200/setup.py#L69-L79
